to.sh

```diff
@@ -17,6 +17,7 @@ qoc() { oc $@ > /dev/null 2>&1; }
 apply() {
   _oc apply -f manifests/ds.yaml
   _oc apply -f manifests/kubelet-configuration-with-swap.yaml
+  wait_for_mcp
```

Comment:
Why should we wait? What is the risk?

Reply:
@fabiand Swap will be provisioned before the flag was applied, and then kubelet won't start

Reply:
@fabiand It can happen that MC will be applied on one node, at the same time KubeletConfig will be applied on another node.

Reply:
Right. But eventually it should fix itself, not?

swap provisioned, kubelet fails .... MC is now rolled out to new node, kubelet (or whole node) restarts, now kubelet is happy and up. Not?

Reply:
@fabiand If kubelet fails - long live MCO on that node

Reply:
@fabiand Recovery is manual and painful, believe me, in order to sync MCO back for that node

Reply:
MCO does not depend on kubelet.
MCO generates ignition for the Machines.

Reply:
@fabiand OK np, you have one controller that reconciles KubeletConfig CR, and another controller that reconciles MachineConfig CR, they are not in sync. If a node got the desired MachineConfig before it gets the desired KubeletConfig, the node becomes NotReady. As the node is NotReady it cannot eventually become Ready, controllers cannot reconcile the node anymore.

Unless I am missing something

Reply:
You are right. If the kubelet is not healthy, then MCs will not be deployed in all cases.

I was only thinking of IPI, where Ignition is used (aka the machine is booted with a cloud init doing the stuff in the MC). In UPI it's different.

Reply:
If we do this change, then please adjust the documentation as well.

Reply:
> You are right. If the kubelet is not healthy, then MCs will not be deployed in all cases.
> 
> I was only thinking of IPI, where Ignition is used (aka the machine is booted with a cloud init doing the stuff in the MC). In UPI it's different.

@fabiand I am not sure I follow wrt IPI vs UPI, does the sync of KubeletConfig and MachineConfig is different in these two methods? as per my understanding the guide as about day-2 installation, when the cluster was already deployed.



> If we do this change, then please adjust the documentation as well.

Sure, will do.